Document OS X installation

We implicitly assume our OS X users already have a compiler and the ffi headers (`xcode-select --install`). This is sometimes an unsafe assumption, especially as our project gets more popular. Let's document how to install on OS X.

The specific error code snippet is below. It fails on the cffi dependency.
``````````````````````
    c/_cffi_backend.c:13:10: fatal error: 'ffi.h' file not found
    
    #include <ffi.h>
    
             ^
    
    1 error generated.
    
    error: command 'cc' failed with exit status 1
`````````````````````````

--- a/docs/installation.rst
+++ b/docs/installation.rst
@@ -128,7 +128,9 @@ build.
 Building cryptography on OS X
 -----------------------------
 
-To install the Xcode command line tools on OS X open a terminal window and run:
+Building cryptography requires the presence of a C compiler and development
+headers. On OS X this is typically provided by Apple's Xcode development tools.
+To install the Xcode command line tools on open a terminal window and run:
 
 .. code-block:: console
 

--- a/docs/installation.rst
+++ b/docs/installation.rst
@@ -125,8 +125,21 @@ You'll also need to generate your own ``openssl.ld`` file. For example::
 You should replace the version string on the first line as appropriate for your
 build.
 
+Building cryptography on OS X
+-----------------------------
+
+To install the Xcode command line tools on OS X open a terminal window and run:
+
+.. code-block:: console
+
+    $ xcode-select --install
+
+This will install a compiler (clang) along with the required development
+headers. If you wish to compile against a more recent OpenSSL than the
+version shipped with OS X see the next section.
+
 Using your own OpenSSL on OS X
-------------------------------
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 To link cryptography against a custom version of OpenSSL you'll need to set
 ``ARCHFLAGS``, ``LDFLAGS``, and ``CFLAGS``. OpenSSL can be installed via

--- a/docs/spelling_wordlist.txt
+++ b/docs/spelling_wordlist.txt
@@ -63,3 +63,4 @@ unpadded
 unpadding
 Verisign
 wildcard
+Xcode

